frontend/src/pages/org/archived-item-qa/ui/replay.ts

```diff
@@ -1,29 +1,73 @@
+import { msg } from "@lit/localize";
+import type { SlRequestCloseEvent } from "@shoelace-style/shoelace";
 import { html } from "lit";
 import { guard } from "lit/directives/guard.js";
 import { when } from "lit/directives/when.js";
 
-import type { ReplayData } from "../types";
+import type { QATab, ReplayData } from "../types";
 
 import { renderSpinner } from "./spinner";
 
+import type { Dialog } from "@/components/ui/dialog";
 import { tw } from "@/utils/tailwind";
 
-export function renderReplay(crawlData: ReplayData) {
+export function renderReplay(crawlData: ReplayData, tab: QATab) {
   return html`
-    <div
-      class=${tw`relative h-full overflow-hidden rounded-b-lg border-x border-b bg-slate-100 p-4 shadow-inner`}
-    >
-      ${guard([crawlData], () =>
-        when(
-          crawlData?.replayUrl,
-          (replayUrl) =>
-            html`<iframe
-              src=${replayUrl}
-              class=${tw`h-full w-full overflow-hidden rounded border bg-neutral-0 shadow-lg`}
-            ></iframe>`,
-          renderSpinner,
-        ),
-      )}
+    <div class="replayContainer ${tw`h-full [contain:paint]`}">
+      <div
+        class=${tw`relative h-full overflow-hidden rounded-b-lg border-x border-b bg-slate-100 p-4 shadow-inner`}
+      >
+        ${guard([crawlData], () =>
+          when(
+            crawlData?.replayUrl,
+            (replayUrl) =>
+              html`<iframe
+                id="interactiveReplayFrame"
+                src=${replayUrl}
+                class=${tw`h-full w-full overflow-hidden rounded border bg-neutral-0 shadow-lg`}
+                @load=${(e: Event) => {
+                  const iframe = e.currentTarget as HTMLIFrameElement;
+
+                  void iframe
+                    .closest(".replayContainer")
+                    ?.querySelector<Dialog>("btrix-dialog.loadingPageDialog")
+                    ?.hide();
+                  /// Prevent anchor tag navigation
+                  iframe.contentDocument?.querySelectorAll("a").forEach((a) => {
+                    a.addEventListener("click", (e: MouseEvent) => {
+                      e.preventDefault();
+                      void iframe
+                        .closest(".replayContainer")
+                        ?.querySelector<Dialog>(
+                          "btrix-dialog.clickPreventedDialog",
+                        )
+                        ?.show();
+                    });
+                  });
+                }}
+              ></iframe>`,
+            renderSpinner,
+          ),
+        )}
+      </div>
+      <btrix-dialog
+        class="loadingPageDialog"
+        .label=${msg("Loading page...")}
+        ?open=${tab === "replay"}
+        no-header
+        @sl-request-close=${(e: SlRequestCloseEvent) => e.preventDefault()}
+      >
+        <sl-progress-bar
```

Comment:
Would be good to have a visible label here, something like:

```html
<h4 class="mb-2">Loading Page...</h4>
```

Good to disable the dialog header though, that makes sense!

Reply:
For the most part the page loads so quickly that the text is a bit distracting, maybe it should be only visible to screenreaders? Thoughts?

https://github.com/webrecorder/browsertrix/assets/4672952/b8ea73fe-a45a-467a-8af3-43f44de102ca

Reply:
Pushed in https://github.com/webrecorder/browsertrix/pull/1775/commits/3fdf7fab75533c72ac8e536a8ffbbb4695eec08f regardless, try it out

Reply:
heh, maybe my internet just isn't that fast 😅 

I liked it when I tried it locally, but seeing it pop in and out so quickly here, your initial instinct to exclude it was probably the right move?  I think not being able to read the text is worse than it taking a while and not knowing _exactly_ what it's doing (loading something.... which people can likely figure out)